src/ng/animate.js

```diff
@@ -416,8 +416,8 @@ var $AnimateProvider = ['$provide', /** @this */ function($provide) {
        * @param {Promise} animationPromise The animation promise that is returned when an animation is started.
        */
       cancel: function(runner) {
-        if (runner.end) {
-          runner.end();
+        if (runner.cancel) {
+          runner.cancel();
```

Comment:
> I thought this change would only align `ng.$animate` with `ngAnimate.$animate`

Isn't it the exact same `$animate` that is being used. Only underlying pieces (e.g. `$$animateQueue`) are different with `ngAnimate`. So, whether we change it or not, `$animate` will be consistent with and without `ngAnimate`.

--
> The reason why I change this code is because otherwise .done isn't called with false, so some switch tests fail.

It sounds more like a test issue, not an implementation issue.

--
> I'd be okay with introducing promise rejection on cancellation as a breaking change for 1.7 on both ng + ngAnimate, but that would mean this perf improvemtn has to wait

Sorry, I didn't get why. Apart from some (bad) tests failing, what is the problem with using `.done()` (but keeping the `runner.end()` call above)?